### PR TITLE
Epic/auth flow

### PR DESCRIPTION
**Pull request by luandro**
Epic solving #33 and #7

### Server
- Add linting with prettier and husk to the server
- Add SSB server 
- Change folder to custom ssb-cashless, so it doesnt conflict with devs ssb
- Remove feedless pub
- Add signup
- Add sendMagicLink with nodemailer
- Add encryption and decryption of secret into and from magic links

### Client
- Add linting to scripts
- Add signup page/form
- Add magic link page
- Add authentication from token on login page

### CODE REVIEW

**Comment by utunga** [comment]
Have we addressed
- [ ] Once I have signed in via email at least once, the toast goes away in future

**Comment by utunga** [comment]
Also, fwiw I found the link that gets sent from ethereal.email 
image

Taking that link http://localhost:3000/login?secret=0e1424139f425dd589c61e6b7d24638a2a7076fbf1915a4a45d510ad8abfa2f3295e51ba40353c020d75520d3c3d6c1e44854d118f61cd675f9effa6ba908d0765803742f606b7fa8b17f6fd3838135e1e613c96f368d5fa3d88de90b257931a5c225b3684681c55018618106decdd2d1827a6986eb05b1de6626904205b64b6548b1a2287474f821724e91588c2d1fa44217de9c1111c1b1e5c2e9341d9bd1b95fc4a+f44e182b93200f3ec503b914e6efd0ed and clicking it, took me to this page


image

Did not log me in?

**Comment by luandro** [comment]
> Have we addressed
> 
> * [ ]  Once I have signed in via email at least once, the toast goes away in future

Yup, the frontend checks if there's token.

That url was wrong, pushed a fix with the correct one.

**Comment by utunga** [comment]
Cool. So I put a sweet ass check mark in the bullet point for "Once I have signed in via email at least once, the toast goes away in future" and i can confirm that the link is fixed now

**Review by utunga — APPROVED** [review]
Giving this an approve so you can merge whenever ... tempted to just merge this as is and handle any other concerns as new issues? basically signifying that Sarah (and you ) should branch of this one from now on ? 

what do you think

File: server/crut/profile.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+//   return out;
+// }
+
+// indexingLog(server);
+
+const spec = {
+  type: 'cashless/profile',
+  props: {
+    name: Overwrite(),
```
[the remark this answers is not in the document]
Comment (luandro) [reply server/crut/profile.js:57]: This is a conversation with need to have, can we leave it for a further issue?

File: server/graphql/resolvers.js [hunk 2]
```diff
@@ -1,83 +1,140 @@
...
+          if (err || !data) reject(err);
+          resolve({
+            ...data.states[0],
+            id: data.key,
+          });
+        });
+      }),
+  },
+  Mutation: {
+    signup: async () => {
```
[the remark this answers is not in the document]
Comment (luandro) [reply server/graphql/resolvers.js]: No, actually this is used on the signup form, so the user's name and description get written to ssb databse.

**Pull request merged**